samples/drivers/counter/alarm/sample.yaml

```diff
@@ -28,12 +38,11 @@ tests:
       - rpi_pico
     integration_platforms:
       - nucleo_f746zg
-    harness_config:
-      type: multi_line
-      ordered: true
-      regex:
-        - "Counter alarm sample"
-        - "Set alarm in 2 sec"
-        - "!!! Alarm !!!"
-        - "Now: [2|3]"
-    depends_on: counter
+  sample.drivers.counter.alarm.stm32_rtc:
+    # This test case is needed because when Timer Counter is available,
+    # RTC Counter will never be selected in the test.
+    # CONFIG_COUNTER_RTC_STM32 flag is enabled by default when RTC is enabled.
+    filter: CONFIG_SOC_FAMILY_STM32 and CONFIG_COUNTER_RTC_STM32
+    extra_configs:
+      # Timer Counter flag needs to be disabled for RTC node to be selected
+      - CONFIG_COUNTER_TIMER_STM32=n
```

Comment:
As I mentioned in the commit message, "this is the only test case that tests STM32 RTC_Alarm currently since its not yet supported in rtc_ll_stm32 driver, & only implemented in counter_ll_stm32_rtc driver." so in my opinion it should be kept in this PR.

Reply:
> We need to find a better way to avoid configuration issues

Of course, will push another PR for this.